### PR TITLE
feature/add merchant coupin cancellation

### PR DESCRIPTION
**Pull request by bolorundurowb**

### CODE REVIEW

File: customer/routes/CoupinRoutes.js [hunk 1]
```diff
@@ -64,7 +64,14 @@ class CoupinRoutes {
       .post(
         PassportMiddleware.verifyJWT1,
         AuthMiddleware.isCustomer,
-        CoupinsCtrl.cancel
+        BookingsCtrl.cancel
+      );
+
+    router.route('/coupin/:coupinId/maerchant-cancel')
+      .post(
+        PassportMiddleware.verifyJWT1,
```
Comment (Saint-lawal) [inline customer/routes/CoupinRoutes.js:72]: A merchant authentication should be used here - "AuthMiddleware.authenticate"

File: customer/routes/CoupinRoutes.js [hunk 2]
```diff
@@ -64,7 +64,14 @@ class CoupinRoutes {
       .post(
         PassportMiddleware.verifyJWT1,
```
Comment (Saint-lawal) [inline customer/routes/CoupinRoutes.js:65]: Same here, i.e. if this is for merchants/admins

**Review by Saint-lawal — CHANGES_REQUESTED** [review]

File: customer/routes/CoupinRoutes.js (hunk 1 above)
Comment (bolorundurowb) [reply customer/routes/CoupinRoutes.js:72]: Yeah, I missed this one

**Comment by bolorundurowb** [comment]
@Saint-lawal made changes to the impl

**Pull request merged**